### PR TITLE
fix: Floating Menu in SQL Left Bar

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
The table and database selector were floating in SQL Left Bar. This solution fixes that by creating a new prop in TableSelector that determines whether to use a created ref or the document body. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:  

image
image

After: 
Screen Shot 2021-03-29 at 5 27 25 PM
Screen Shot 2021-03-29 at 5 27 29 PM

### TEST PLAN
This was visually tested. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:#13842
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by AAfghahi** [comment]
@junlincc and oh weird I can't tag yousoph. :/

**Review by eschutho — CHANGES_REQUESTED** [review]
I think we can accomplish this with overflow: visible in a few places.

**Comment by ktmud** [comment]
I also noticed that the menu overlaps with SQL Editor control buttons when table names are very long. Can you make sure this is also fixed?

messed-up

**Comment by AAfghahi** [comment]
@ktmud 

Oh wow that looks weird. Thanks for bringing it up. I will need to find some databases that match that length and test.

**Comment by etr2460** [comment]
We ran into the bug this is fixing in our weekly release today. Since this PR is still in progress and the PR that introduced the bug reverts cleanly (and easily) how do we feel about reverting https://github.com/apache/superset/pull/13694?

cc @AAfghahi @yousoph @junlincc

**Comment by AAfghahi** [comment]
@etr2460 that sounds good to me, this should be merged soon.

**Comment by AAfghahi** [comment]
> I also noticed that the menu overlaps with SQL Editor control buttons when table names are very long. Can you make sure this is also fixed?
> 
> messed-up

Hello @ktmud how does this look: 
Screen Shot 2021-04-01 at 5 42 09 PM
You could still use the tooltip to see the full name.

**Comment by eschutho** [comment]
This looks good @AAfghahi. Were you able to get the ellipses to show up? I didn't see them.

**Comment by AAfghahi** [comment]
@eschutho no, I think I need to set a definable width.

**Comment by eschutho** [comment]
I don't have a huge preference whether we have the ellipsis or not.. @steejay wdyt?

**Comment by Steejay** [comment]
@AAfghahi @eschutho hello! Ideally we'd add an ellipsis so that the user knows that the copy is cut off. This will then help indicate that a tooltip would reveal the complete copy.

**Comment by AAfghahi** [comment]
https://user-images.githubusercontent.com/48933336/113441862-ffd1b400-93bc-11eb-8e59-469609a5a4df.mov

I think this is everything that we wanted.

**Comment by ktmud** [comment]
It'd be really appreciated if the menu can be expanded to fullwidth and show full table names without tooltips.

We had users writing thank you notes when we first made it possible.


On another note, can we also flatten the round corners here:

round corner

**Comment by AAfghahi** [comment]
@ktmud I can definitely change the rounded corners and the caret change. For the expanded table names, we could also enable overflow-x so that there is a scroll inside the menu? @Steejay what do you think?

**Comment by Steejay** [comment]
@ktmud @AAfghahi i dont think we want to break the boundaries of the dropdown like the above screenshot. it looks broken in this scenario. If there are key identifiers at the end of the title I wonder if a middle ellipsis works better? sim to how the mac finder displays a list.

**Comment by AAfghahi** [comment]
@ktmud 
Screen Shot 2021-04-05 at 11 15 31 AM
done :)

**Review by eschutho — APPROVED** [review]
LGTM. Thanks @AAfghahi!
Needs visual testing still.

**Comment by ktmud** [comment]
@Steejay I think it's find for the dropdown menu to be wider than the select box (it shouldn't be shorter than it though).

Even AntD has a `dropdownMatchSelectWidth` option which you can set to `false` to allow this auto-expand:
https://codesandbox.io/s/search-with-sort-antd4150-forked-nkude?file=/index.js

@AAfghahi could you add screenshots of how the select look like in Explore page chart controls and Dashboard FilterBox just for confirmation?

**Comment by AAfghahi** [comment]
@ktmud I don't have any filter boxes with really long names. 
Screen Shot 2021-04-05 at 1 02 46 PM

Screen Shot 2021-04-05 at 1 12 12 PM

Is there a public database that does so I can check it?

**Comment by AAfghahi** [comment]
Screen Shot 2021-04-05 at 1 52 38 PM


ok found one

**Comment by betodealmeida** [comment]
> @ktmud I don't have any filter boxes with really long names.
> Screen Shot 2021-04-05 at 1 02 46 PM
> 
> Screen Shot 2021-04-05 at 1 12 12 PM
> 
> Is there a public database that does so I can check it?

You can just add a row. Go to SQL Lab and run something like this:

```sql
UPDATE table SET country = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX' WHERE country = 'Angola';
```

**Comment by AAfghahi** [comment]
@ktmud 
Screen Shot 2021-04-05 at 3 36 35 PM
Screen Shot 2021-04-05 at 3 35 30 PM
Screen Shot 2021-04-05 at 3 36 24 PM
(It is cool that it tells you how long it will be)

**Comment by ktmud** [comment]
@AAfghahi Thank you for the screenshots!

**Comment by eschutho** [comment]
@yousoph did visual testing
🏷 ready-to-merge

**Review by betodealmeida — APPROVED** [review]
LGTM

**Comment by maltoze** [comment]
image

🤣

File: superset-frontend/src/SqlLab/main.less [hunk 1]
```diff
@@ -300,6 +300,7 @@ div.Workspace {
   .schemaPane-enter-done,
   .schemaPane-exit {
     transform: translateX(0);
+    z-index: 1020;
```
Comment (maltoze) [inline superset-frontend/src/SqlLab/main.less:303]: image

Comment (AAfghahi) [reply superset-frontend/src/SqlLab/main.less:303]: https://github.com/apache/superset/pull/13972

**Pull request merged**